### PR TITLE
Fix version scheme

### PR DESCRIPTION
**Pull request by twometresteve**
This change adopts the following version pattern when publishing for CI (or locally, should you wish to us it):

<local version>-<ci flag>-<branch name>.<cmomit id>

Where:
- Local version is retrieved from `lerna.json` (x.y.x only)
- "ci-' is added to distinguish from local use and avoid publishing clashes

### CODE REVIEW

File: scripts/common.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    determineVersion: function determineVersion() {
+        // Form version string - branch name and commit id
+        let commitId = this.getCommitId();
+        let branchName;
+        if (process.env.BRANCH_NAME) {
+            console.log('Using BRANCH_NAME from the environment')
+            branchName = process.env.BRANCH_NAME
+        } else {
+            console.log('Using local Git repo to determine branch name (environment BRANCH_NAME may also be used)')
+            branchName = this.run('git branch --show-current').replace(/\//g,"-")
```
Comment (bengourley) [inline scripts/common.js:19]: this seems like a new feature of git that my version doesn't have. Can you use `git rev-parse --abbrev-ref HEAD` instead for wider compatibility?

File: scripts/common.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        }
+
+        // Distinguish from local use
+        let ciIndicator = (process.env.BUILDKITE ? "ci-" : "")
+
+        // Get the current version from lerna.json
+        const lerna = require("../lerna.json");
```
Comment (bengourley) [inline scripts/common.js:26]: `require` should be at the top of the file, though the linter may tell you that when it is enabled

File: scripts/common.js (hunk 1 above)
Comment (bengourley) [inline scripts/common.js:19]: are there any other characters than `/` that are allowed in branch names that would break this?

Comment (twometresteve) [reply scripts/common.js:19]: There might be others, but I don't think we're in the habit of using them and `/` is the only character that the Buildkite agents replace when creating the `BRANCH_NAME` environment variable.  So I went with the same approach primarily for consistency.

File: scripts/common.js (hunk 2 above)
Comment (twometresteve) [reply scripts/common.js:26]: Moved.

File: scripts/common.js (hunk 1 above)
Comment (twometresteve) [reply scripts/common.js:19]: Of course.

**Comment by twometresteve** [comment]
> Doesn't seem like the new files are linted on CI. I think they need to be added to the ci dockerfile `COPY scripts scripts`. I can run `npm run test:lint` locally to see the lint errors, or configure your editor's eslint integration.
 
Now added, IDE set up and violations resolved.

File: scripts/common.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    if (process.env.BRANCH_NAME) {
+      console.log('Using BRANCH_NAME from the environment')
+      branchName = process.env.BRANCH_NAME
+    } else {
+      console.log('Using local Git repo to determine branch name (environment BRANCH_NAME may also be used)')
+      branchName = this.run('git rev-parse --abbrev-ref HEAD').replace(/\//g, '-')
+    }
+
+    // Distinguish from local use
+    const ciIndicator = (process.env.BUILDKITE ? 'ci-' : '')
```
Comment (Cawllec) [inline scripts/common.js:24]: The `BUILDKITE` environment variable will need to be added into the `docker-compose.yml` in order for it to be passed through to the container.

Comment (twometresteve) [reply scripts/common.js:24]: Nice spot - I added that at the last minute and assumed too much.

**Pull request merged**